### ISSUE TITLE
New provides matcher syntax

### ISSUE THREAD

**Issue by blakechambers** [blakechambers/serial-spec#1]
```ruby

require 'spec_helper'

RSpec.describe "index" do
  include SerialSpec

  collection_name "posts"
  resource_class Post
  resource_scope Post.all

  with_request "GET /#{collection_name}.json" do

    context 'when post exists' do
      let!(:existing_post) { resource_scope.make! }
      
      it "should provide the post" do
        perform_request!

        # new syntax
        expect(body[:posts].first.execute).to provide(existing_post, as: PostSerializer)
        expect(body[:posts].execute).to provide([existing_post], as: PostSerializer)
        
        # if type == SerialSpec::ParsedBody, call .execute, else hash
        expect(body[:posts]).to provide([existing_post], as: PostSerializer)
        expect(body[:posts]).to provide([existing_post], as: PostSerializer)    
      end
    end

  end
end

```

* `as` should be optional, when specified it forces that serializer to be used.  Without it, the existing post should call `as_json` to perform the hash comparison to body.

**Comment by blasterpal** [comment]
- [ ] - Detect ActiveSupport Serializers and load Provides if present
- [ ] - Use Appraisals to run test with and without ActiveModel Serializers .08

**Comment by blakechambers** [comment]
closed by #2.
